### DISCUSSION TITLE
Usage

### DISCUSSION THREAD

**Discussion by ghost** [lingfish/mmng-ui#2] [General]
I might be dumb, but I am using rx_fm on a HackRF, I can see the standard output which I can pipe into multimon-ng, but how do I get that output into mmng-ui? Should I be using some other tool to send output to port 8888?

**Comment by lingfish** [comment]
Heya, not dumb at all!

I've not used `rx_fm`, but see it's a clone of `rtl_fm`.  Both tools will output samples to a file, or standard output (and not the network), so you'll need `nc` (netcat) in the middle.

It also looks like `rx_fm` defaults to a sample rate of 24k, where `multimon-ng` wants 22050Hz.

So the chain of tools would look like this:

```mermaid
graph LR;
    rx_fm-->stdout --> netcat -->|UDP port x| mmng-ui --> A[multimon-ng run internally by mmng-ui];
```

Thus, a commandline might look like this (note the trailing dash before the pipe to `nc`!):

```bash
rx_fm -f 144.640M -s 22050 <other options> - | nc -u localhost 8888
```

Obviously run this after starting `mmng-ui`.

Hope this helps, let me know either way!

**Comment by lingfish** [comment]
Hi @Arkwin, did this help?  All good?

**Reply by ghost** [reply]
Hey! Thanks. It helped some! I got it working, and by that I mean sending data to it. I just cannot seem to properly decode FLEX in my area.

**Reply by lingfish** [reply]
Good to hear -- do you mean in general, or specific to this project?

I don't even think FLEX is used in my area (Australia) as far as I know.

**Reply by ghost** [reply]
In general. I am using a HackRF and using rx_fm and I can pipe that nc to mmng-ui, but even without mmng-ui, I just get FLEX messages that are too garbled even after attempting correction. :(

**Reply by ghost** [reply]
For reference, this is all I get in multimon-ng.
![image](https://github.com/user-attachments/assets/6e92ee15-5b4e-47f6-971b-a77c891814a5)

**Reply by lingfish** [reply]
Hrm, yeah, dunno.  I see a fair few issues at the `multimon-ng` repo around FLEX and 3200 baud, amongst others.

Have you tried any of the samples at [sigid](https://www.sigidwiki.com/wiki/FLEX)?  I just tried both 1600 bps (success) and 3200 (epic fail).

**Comment by ghost** [comment]
I have not gotten that far,  but did see the 3200 baud issues on the multimon-ng repo. Thank you for your help though! I will mess around with it some more. :)


________________________________

**Reply by lingfish** [reply]
Yeah, I just tried with their 3200 sample and a recent build of `multimon-ng`, and it seems to mis-detect as 1600 baud and of course, that'd be a jumbled mess:

```shell
multimon-ng -a FLEX -t mp3 /tmp/Flex_3200.mp3 -v 3
multimon-ng 1.3.1
  (C) 1996/1997 by Tom Sailer HB9JNX/AE4WA
  (C) 2012-2024 by Elias Oenal
Available demodulators: POCSAG512 POCSAG1200 POCSAG2400 FLEX FLEX_NEXT EAS UFSK1200 CLIPFSK FMSFSK AFSK1200 AFSK2400 AFSK2400_2 AFSK2400_3 HAPN4800 FSK9600 DTMF ZVEI1 ZVEI2 ZVEI3 DZVEI PZVEI E
EA EIA CCIR MORSE_CW DUMPCSV X10 SCOPE
Enabled demodulators: FLEX
FLEX: Locked
FLEX: SyncInfoWord: sync_code=0xb068 baud=1600 levels=4 polarity=NEG zero=0.029322 envelope=0.125693 symrate=1600.403226
FLEX: State: FIW
FLEX: FrameInfoWord: cycleno=08 frameno=117 fix3=0x00 time=35:39 offset=0.00:00:00
FLEX: State: SYNC2
FLEX: State: DATA
FLEX: Phase A Fixed 2 errors @ 0x00800001  (0x4bc05407 -> 0x4b405406)
FLEX: Phase A Fixed 1 errors @ 0x00200000  (0x2d625967 -> 0x2d425967)
FLEX: Phase A Fixed 2 errors @ 0x00220000  (0x3ff2cfb3 -> 0x3fd0cfb3)
FLEX: Phase A Data corruption - Unable to fix errors.
FLEX: Garbled message at block 3
FLEX: Phase B Data corruption - Unable to fix errors.
FLEX: Garbled message at block 0
FLEX: State: SYNC1
FLEX: SyncInfoWord: sync_code=0xb068 baud=1600 levels=4 polarity=NEG zero=0.034264 envelope=0.124727 symrate=1600.037794
FLEX: State: FIW
FLEX: FrameInfoWord: cycleno=08 frameno=118 fix3=0x00 time=35:41 offset=0.00:00:02
FLEX: State: SYNC2
FLEX: State: DATA
FLEX: Phase A Fixed 2 errors @ 0x00020004  (0x6bc04c0f -> 0x6bc24c0b)
FLEX: Phase A Fixed 2 errors @ 0x02000008  (0x7ec2dd2b -> 0x7cc2dd23)
FLEX: Phase A Data corruption - Unable to fix errors.
FLEX: Garbled message at block 2
FLEX: Phase B Fixed 2 errors @ 0x20100000  (0x1f701402 -> 0x3f601402)
FLEX: Phase B Data corruption - Unable to fix errors.
FLEX: Garbled message at block 1
```
